src/widgets/component-new-edit/components/variables/external-variables.jsx

```diff
@@ -41,7 +44,17 @@ export const defaultProps = {
 
 // Component
 
-function ExternalVariables({ formName, selectorPath, errors, addErrors }) {
+function ExternalVariables({ formName, selectorPath, errors, addErrors, componentsStore }) {
+  const scopeOption = Object.values(componentsStore)
+  .filter(component=> component.type === "LOOP" && !component.basedOn || component.type === "LOOP" && component.basedOn && componentsStore[component.basedOn] && componentsStore[component.basedOn].type != "LOOP" )
```

Comment:
Same comment as before 👆
If this is exactly the same code, we can maybe extract into a method, that you'll be able to unit test.